### DISCUSSION TITLE
unable to start MTB after migrating to new laptop

### DISCUSSION THREAD

**Discussion by Frans2222** [mytourbook/mytourbook#1237] [Q&A]
Hi all, installed MTB 23.10 on my new laptop and copied the derby database. Unfortunately I deleted the files on the old one. Now I receive the attached errors.
![error2](https://github.com/mytourbook/mytourbook/assets/109601438/7acbec54-1d8e-49f5-93ee-82dde2488e7f)
![error1](https://github.com/mytourbook/mytourbook/assets/109601438/fef922f7-7e58-49c0-a287-3c4e2710856d)

Btw, I do have backups when new versions were introduced, but I'm unable to reinstall these as well. Please help.

**Comment by Frans2222** [comment]
Added the log

[derby.log](https://github.com/mytourbook/mytourbook/files/13702934/derby.log)

**Comment by FJBDev** [comment]
@Frans2222 Not sure what the errors in the log mean....

Can you try to launch MT with no database at all to see if you can start it successfully ?

**Reply by Frans2222** [reply]
yes, no problem

**Reply by FJBDev** [reply]
Is the database version that you're trying to use  from 23.10 ? if not, which version was it ?

**Reply by Frans2222** [reply]
it was 23.10. when i add the .metadata it works, adding the derby on top of it, it fails.

**Reply by Frans2222** [reply]
New SQL error: The dir does not contain the expected 'service properties'file. The derby is over 45GB in seize, could that be an issue?

**Reply by wolfgang-ch** [reply]
Can you replace the tour database again from a backup.

* Close MT
* Delete the folder `C:\Users\<YOUR NAME>\mytourbook\derby-database`
* Copy the **COMPLETE** folder `...derby-database` from your backup into `C:\Users\<YOUR NAME>\mytourbook\derby-database`
* Restart MT

I'm doing this procedure very often to copy my production tour db into my dev environment by using this script

```
rmdir workspace\derby-database /q/s
xcopy "C:\Users\Wolfgang\mytourbook\derby-database" workspace\derby-database\ /S
```

**Comment by Frans2222** [comment]
Thanks Wolfgang.

Followed your instructions. It keeps asking for the service.properties file. However, that file is included. I don't get it.
     On Tuesday, December 19, 2023 at 10:24:50 AM GMT+1, Wolfgang Schramm ***@***.***> wrote:  
 
 


Can you replace the tour database again from a backup.
   
   - Close MT
   - Delete the folder C:\Users\<YOUR NAME>\mytourbook\derby-database
   - Copy the COMPLETE folder ...derby-database from your backup into C:\Users\<YOUR NAME>\mytourbook\derby-database
   - Restart MT

I'm doing this procedure very often to copy my production tour db into my dev environment by using this script
rmdir workspace\derby-database /q/sxcopy "C:\Users\Wolfgang\mytourbook\derby-database" workspace\derby-database\ /S

**Reply by wolfgang-ch** [reply]
>It keeps asking for the service.properties file. 

Where is it asking this file ?

Sometimes it can be also helpful to ask the internet because the derby SQL database is widely used and each error has an unique error code


I have this file also in my backups

![image](https://github.com/mytourbook/mytourbook/assets/1283445/a73997b3-238c-4233-bde9-2f20770affca)

**Comment by Frans2222** [comment]
Hi Wolfgang, at somepoint a date was added to the files in my backup dirs after installing the 2023.10 version. I copied the .dat files from the seg0 dir just for screenshot1.

I've tried everything, like installing old versions, removing the date from hundreds of files etc but with my limited knowledge was unable to restore the database.

Never mind, I was able to recover with Easeus an old backup file from the old laptop and only lost a few weeks of data. The total derby-database is just over 300MB now, where the corrupted one was 46GB (screenshot2).

Thanks for your support.

Kind regards,
Frans

     On Wednesday, December 20, 2023 at 07:50:02 AM GMT+1, Wolfgang Schramm ***@***.***> wrote:  
 
 



It keeps asking for the service.properties file.


Where is it asking this file ?

Sometimes it can be also helpful to ask the internet because the derby SQL database is widely used.

I have this file also in my backups

image.png (view on web)

**Comment by FJBDev** [comment]
> The total derby-database is just over 300MB now, where the corrupted one was 46GB (screenshot2). Thanks for your support. Kind regards, Frans 

Glad to read that you were able to make it work with another backup.
I think the size of 45Gb is definitely a red herring and shows that something went wrong during the original backup since there is no way only a couple of tours take 40Gb +.

What ive done for the last 4 years is zip the derbydatabase folder and ive had 0 issues restoring backups.

**Reply by wolfgang-ch** [reply]
I' also zipping the derby database folder for the backup because restoring a single database file do not work

Searching for "xsdb2 derby" (error code in the first screenshot) finds this interresting document https://givemethechills.com/2015/05/24/recovering-a-corrupted-embedded-apache-derby-database-after-an-error-xj040-or-if-you-get-an-error-xsdg2/

**Comment by Frans2222** [comment]
Very interesting indeed! Thanks guys.
     On Monday, December 25, 2023 at 09:23:10 AM GMT+1, Wolfgang Schramm ***@***.***> wrote:  
 
 


I' also zipping the derby database folder because restoring a single database file do not work

Searching for "xsdb2 derby" (error code in the first screenshot) finds this interresting document https://givemethechills.com/2015/05/24/recovering-a-corrupted-embedded-apache-derby-database-after-an-error-xj040-or-if-you-get-an-error-xsdg2/

**Comment by DataWizard0103** [comment]
After installing version 2023.10, I encountered an issue where dates were added to files in my backup directories. This made the restoration process difficult despite trying several solutions, including installing older versions and renaming files manually. As I am not an expert in this field, I was unable to recover the database successfully. However, I was able to recover most of the data by using data recovery professional software to retrieve an older backup file from my previous laptop. I am attaching screenshot below for your reference. 
 
![stellar-data-recovery-professional-31091-1](https://github.com/mytourbook/mytourbook/assets/160112115/8b1c6604-85cb-42f9-8fd3-d0d969922a40)
